doc/porting_processing.dox

```diff
@@ -0,0 +1,96 @@
+
+    - ParameterBoolean has been replaced with QgsProcessingParameterBoolean.
+      - Retrieving the parameter value should be done with self.parameterAsBool( parameters, PARAM_NAME, context). This will return a True or False value depending on the input parameter.
+    
+    - ParameterCrs has been replaced with QgsProcessingParameterCrs.
+      - Retrieving the parameter value should be done with self.parameterAsCrs( parameters, PARAM_NAME, context). CRS parameters are returned as QgsCoordinateReferenceSystem objects. In 2.x crs parameters were returned as a string, leaving the algorithm
+responsible for interpreting this string and working out how to convert it to a suitable CRS object. In 3.0 this is automatically handled
+and a QgsCoordinateReferenceSystem object is returned, ready for use. If you require a string version of the CRS, use the
+QgsCoordinateReferenceSystem methods like authid() to obtain a string representation of the CRS in the desired format.
+
+    - ParameterExtent has been replaced with QgsProcessingParameterExtent.
+      - Retrieving the parameter value should be done with self.parameterAsExtent( parameters, PARAM_NAME, context). Extent parameters are returned as QgsRectangle. In 2.x extent parameters were returned as a delimited string, leaving the algorithm
+responsible for parsing the string and validating it. In 3.0 this is automatically handled and a QgsRectangle returned, ready for use.
+
+
+- Use featuresources and sinks wherever possible
+
+Update outputs
+
+ - OutputNumber has been replaced with QgsProcessingOutputNumber. QgsProcessingOutputNumber is a drop-in replacement for OutputNumber.
+ - OutputString has been replaced with QgsProcessingOutputString. QgsProcessingOutputString is a drop-in replacement for OutputString.
+ 
+- OutputVector -> 
+
+- processAlgorithm has a new signature....
+
+Use parameterAsSink instead of getOutputFromName
+
+- Since processing algorithms can now be run in the background, it's important to implement support for cancelation in your
+algorithms. Whenever looping (or before/after calling lengthy operations) listen out for user cancelation via the provided
+feedback object. E.g.
+
+   if feedback.isCanceled():
+       break
+
+- Give feedback via the feedback object, not directly to QgsMessageLog. E.g.
+
+   feedback.pushInfo('Feature {} was missing geometry, skipping'.format(f.id())
+   feedback.reportError('Input layer CRS is not valid, cannot reproject')
+
+- Use exceptions only for FATAL errors which force a model to terminate. Algorithms should handle common cases
+such as having no features in an input layer without throwing exceptions (instead, an empty layer should be output). This
+allows more flexibility for users creating models. They may have created models which "route" features to different algorithms
```

Comment:
Should this be "this allow FOR more flexibility"? Sounds strange to me as it is currently, but I'm not a native speaker.